src/editor/classes/EditorStatusEvents.js

```diff
@@ -48,29 +78,23 @@ class EditorEventHandler {
     }
 
     onCursorActivity(proxy, editor) {
-        var out = [];
-        var sels = editor.listSelections();
-        for (var i = 0; i < sels.length; i++) {
-            out[i] = {  anchor: {
-                            line: sels[i].anchor.line,
-                            ch: sels[i].anchor.ch
-                        },
-                        head: {
-                            line: sels[i].head.line,
-                            ch: sels[i].head.ch
-                        }
-                    };
-        }
-        proxy.setValue("selections", out);
-        proxy.setValue("selectionsText", editor.getSelections("\n"));
-        var cur = editor.getCursor();
-        proxy.setValue("cursor", [cur.line, cur.ch]);
+        clearTimeout(this.cursorActivityTimer);
+        // We put this in a small timer so we don't flood the queue
+        // during selection and fast cursor movement.
+        this.cursorActivityTimer = setTimeout(function() {
+            proxy.sendEditorEvent("J_EVT_CURSOR_ACTIVITY", this.cursorActivityObject(editor));
+        }.bind(this), 20);
```

Comment:
I'm javascript retarded... the fact that I've managed to do as much as I have(eg. CodeMirror plugin I wrote forever ago) is a miracle... I'll change this real quick.